src/Microsoft.AspNet.Server.Kestrel/Infrastructure/KestrelThread.cs

```diff
@@ -262,11 +267,22 @@ private void ThreadStart(object parameter)
 
         private void OnPost()
         {
-            DoPostWork();
-            DoPostCloseHandle();
+            var loopsRemaining = _maxLoops;
+            bool wasWork;
+            do
+            {
+                // Queues swapped
+                wasWork = DoPostWork();
+                wasWork = DoPostCloseHandle() || wasWork;
```

Comment:
I don't think coalescing is an issue since we always call `uv_async_send` after enqueing to `_workAdding`. We've verified that calling `uv_async_send` will not coalesce if it is the first call after the post callback *starts*.